### ISSUE TITLE
Need poetry Add some_package --lock --no-update

### ISSUE THREAD

**Issue by sshankar619** [python-poetry/poetry#7903]
- **Poetry version**: 
- **Python version**: 
- **OS version and name**: 
- **pyproject.toml**: 

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

Basically , we can add a package  in pyproject.toml file manually and perform poetry lock --no-update.
Need to do the same programatically by poetry add some_package --lock --no-update
To reduce the poetry locking time

Or need clarity if poetry add some_package --lock  itself will work like --no-update  ( i  cant see this detail anywhere in website)
